nengo_loihi/splitter.py

```diff
@@ -4,184 +4,280 @@
+                        "configured as on_chip." % (pre, conn)
+                    )
+                self.host_objs.add(pre)
+                self.chip_objs.remove(pre)
+
+    def _place_probes(self, network):
+        """Place probes. Probes go where their probed object is."""
+
+        for probe in network.all_probes:
+            obj = base_obj(probe.target)
+            if obj in self.host_objs:
+                self.host_objs.add(probe)
+            elif obj in self.chip_objs:
+                self.chip_objs.add(probe)
+            else:
+                raise BuildError("Object (%r) is not a part of the network" % (obj,))
 
     def on_chip(self, obj):
-        if isinstance(obj, Probe):
-            obj = base_obj(obj.target)
-        if not isinstance(obj, (Ensemble, Node)):
+        if not isinstance(obj, (Ensemble, Node, Probe)):
             raise BuildError(
                 "Locations are only established for ensembles ",
                 "nodes, and probes -- not for %r" % (obj,),
             )
-        if obj not in self._seen_objects:
-            raise BuildError("Object (%r) is not a part of the network" % (obj,))
-        return obj in self._chip_objects
+        if obj in self.chip_objs:
+            return True
+        elif obj in self.host_objs:
+            return False
+        raise BuildError("Object (%r) is not a part of the network" % (obj,))
+
+
+class Split:
+    """Creates a set of directives to guide the builder.
+
+    Parameters
+    ----------
+    network : Network
+        The original Network supplied to the builder.
+    precompute : bool, optional (Default: None)
+        Whether model inputs should be precomputed to speed up simulation.
+        The splitter will determine precomputable objects regardless,
+        but if
+
+
+    Attributes
+    ----------
+
+    """
+
+    def __init__(self, network, precompute=None, remove_passthrough=True):
+        self.network = network
+
+        # Place objects on host or chip
+        self.hostchip = HostChipSplit(network)
+
+        # Determine how passthrough nodes will be handled
+        if remove_passthrough:
+            passthroughs = set(obj for obj in network.all_nodes if is_passthrough(obj))
+            ignore = self.hostchip.host_objs - passthroughs
+            self.passthrough = PassthroughSplit(network, ignore)
+        else:
+            self.passthrough = PassthroughSplit(None)
+
+        # Determine which host objects are precomputable
+        self._precomputable = PrecomputableSplit(
+            network, self.hostchip, self.passthrough, strict=precompute is True
+        )
+        self.precompute = precompute
+        if self.precompute is None:
+            self.precompute = self.precomputable()
+
+    def precomputable(self, obj=None):
+        return self._precomputable.precomputable(obj=obj)
+
+    def on_chip(self, obj):
+        return self.hostchip.on_chip(obj)
```

Comment:
I think it would be great to have @arvoelke look at these splitter changes, because I think he wrote most of the original code that's being changed.

Reply:
Done :)